### PR TITLE
Expand batch plots to support processing by level

### PR DESCRIPTION
**Pull request by EdwardSafford-NOAA**
## Description

Expand batch plots to handle levels like channels.  Step through levels, creating data subsets by level, and plotting each level.  This is an example of the output (note the holes in the data are actually missing data files and are not errors):
ompsnp_npp 1 omg

@CoryMartin-NOAA I'm not sure if this partially answers #124.

## Dependencies

None

## Impact

None

Completes #156 .

### CODE REVIEW

File: src/eva/plotting/emcpy/plot_tools/figure_driver.py [hunk 1]
```diff
@@ -65,9 +65,21 @@ def figure_driver(config, data_collections, timing, logger):
...
             channels_str_or_list = batch_conf.get('channels', [])
             channels = parse_channel_list(channels_str_or_list, logger)
+            step_vars = channels
+            step_var_name = 'channel'
+            title_fill = ' Ch. '
+
+            # Get list of levels
+            levels_str_or_list = batch_conf.get('levels', [])
+            levels = parse_channel_list(levels_str_or_list, logger)
+            if levels != []:
```
Comment (kevindougherty-noaa) [inline src/eva/plotting/emcpy/plot_tools/figure_driver.py]: I believe the more "pythonic" way of addressing this conditional would be:

`if levels:`

See here: https://stackoverflow.com/questions/53513/how-do-i-check-if-a-list-is-empty

File: src/eva/plotting/emcpy/plot_tools/figure_driver.py [hunk 2]
```diff
@@ -65,9 +65,21 @@ def figure_driver(config, data_collections, timing, logger):
...
+                step_vars = levels
+                step_var_name = 'level'
+                title_fill = ' Lv. '
 
             # Set some fake values to ensure the loops are entered
             if variables == []:
```
Comment (kevindougherty-noaa) [inline src/eva/plotting/emcpy/plot_tools/figure_driver.py]: Same comment as above.

**Review by kevindougherty-noaa — COMMENTED** [review]
Just a small change request in regards to a conditional.

File: src/eva/plotting/emcpy/plot_tools/figure_driver.py (hunk 1 above)
Comment (EdwardSafford-NOAA) [reply src/eva/plotting/emcpy/plot_tools/figure_driver.py]: Good to know, thanks.  Change made.

File: src/eva/plotting/emcpy/plot_tools/figure_driver.py (hunk 2 above)
Comment (EdwardSafford-NOAA) [reply src/eva/plotting/emcpy/plot_tools/figure_driver.py]: Change made here and elsewhere in the (previously) unchanged code.

**Review by kevindougherty-noaa — APPROVED** [review]
Looks good by me, thanks for making the changes. @CoryMartin-NOAA is out of office until Monday.

**Comment by EdwardSafford-NOAA** [comment]
I note the application and notebook tests are failing because not all required elements (like matplotlib) are not loading on the test platform.  That doesn't seem like it might be something these changes brought about.

**Comment by EdwardSafford-NOAA** [comment]
I found a problem with two of the other test scripts as a result of these changes.  I'm going to shift this PR to draft until I figure that out.

**Comment by EdwardSafford-NOAA** [comment]
Found and fixed the problem.  I had missed a necessary default if neither channels nor levels are in play.  Got lucky that I found the problem so soon.

**Comment by CoryMartin-NOAA** [comment]
I think the test failures now are unrelated (ugh again) but will double check by rerunning an old action

File: src/eva/data/data_collections.py [hunk 3]
```diff
@@ -188,31 +190,47 @@ def get_variable_data_array(self, collection_name, group_name, variable_name, ch
...
+                                      f'Level is not a dimension in Dataset')
+                # Make sure it is a list
+                levels_sel = []
+                levels_sel.append(levels)
+
+                # Create a new DataArray with the requested channels
+                data_array_levels = data_array.sel(Level=levels_sel)
+                return data_array_levels
 
         else:
```
Comment (CoryMartin-NOAA) [inline src/eva/data/data_collections.py]: I think this else needs modified now, right? Or at very least the error message that gets written out

File: src/eva/plotting/emcpy/plot_tools/figure_driver.py [hunk 4]
```diff
@@ -65,27 +65,40 @@ def figure_driver(config, data_collections, timing, logger):
...
+            step_var_name = 'channel'
+            title_fill = ' Ch. '
+
+            # Get list of levels, conditionally override step variables
+            levels_str_or_list = batch_conf.get('levels', [])
+            levels = parse_channel_list(levels_str_or_list, logger)
+            if levels:
+                step_vars = levels
+                step_var_name = 'level'
+                title_fill = ' Lv. '
```
Comment (CoryMartin-NOAA) [inline src/eva/plotting/emcpy/plot_tools/figure_driver.py]: personal nit pick, would prefer `' Lev. '` , I think the one extra letter makes it less ambiguous, but curious what others think.

**Review by CoryMartin-NOAA — COMMENTED** [review]
Thanks for this, this will be a fantastic and really useful addition. I have just a couple comments.

**Comment by CoryMartin-NOAA** [comment]
Regarding #124 I'd have to try it, but theoretically, yes, this should do it.

File: src/eva/plotting/emcpy/plot_tools/figure_driver.py (hunk 4 above)
Comment (EdwardSafford-NOAA) [reply src/eva/plotting/emcpy/plot_tools/figure_driver.py]: Done.

File: src/eva/data/data_collections.py (hunk 3 above)
Comment (EdwardSafford-NOAA) [reply src/eva/data/data_collections.py]: Fair point.  See if these changes handle things better.

**Review by CoryMartin-NOAA — APPROVED** [review]

File: src/eva/data/data_collections.py (hunk 3 above)
Comment (CoryMartin-NOAA) [reply src/eva/data/data_collections.py]: Yep thanks!

**Comment by CoryMartin-NOAA** [comment]
Since this touches the data collections, I would like either @danholdaway or @asewnath to approve before merging in.

**Review by asewnath — APPROVED** [review]

**Review by danholdaway — APPROVED** [review]

**Pull request merged**